runtime/lua/vim/lsp/protocol.lua

```diff
@@ -923,4 +923,330 @@ function protocol.resolve_capabilities(server_capabilities)
   return server_capabilities
 end
 
+--Lsp methods keep at last
+protocol.Methods = {
+  --A request to resolve the implementation locations of a symbol at a given text
+  --document position. The request's parameter is of type [TextDocumentPositionParams]
+  --(#TextDocumentPositionParams) the response is of type {@link Definition} or a
+  --Thenable that resolves to such.
+  --@param ImplementationParams
+  Implementation = 'textDocument/implementation',
+  --A request to resolve the type definition locations of a symbol at a given text
+  --document position. The request's parameter is of type [TextDocumentPositionParams]
+  --(#TextDocumentPositionParams) the response is of type {@link Definition} or a
+  --Thenable that resolves to such.
+  --@param TypeDefinitionParams
```

Comment:
I think we use `---` wherever we intend machine-readable docstrings.

```suggestion
  ---@param TypeDefinitionParams
```

Reply:
before tried use `---` but lua_ls seems like not parse this for field.

Reply:
? we use that everywhere:

    git grep -P '\---@param'

Reply:
![image](https://github.com/neovim/neovim/assets/41671631/e947212e-23b3-4eab-b2c6-c0a2da5452bb)

Reply:
I think `@param` here is in any case not appropriate given that this is a string constant, and not a function. Strings don't have parameters.

I'm also not sure we want the text at all 1:1 as it is in the JSON. The `Thenable` stuff is nonsense in our context and the links don't work as is.

Reply:
yep when i write also think it does not make sense.